### PR TITLE
Lambda docker deployment

### PR DESCRIPTION
**Pull request by acodeninja**

### CODE REVIEW

File: modules/lambda-example/deploy.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module "lambda-with-container" {
+  source      = "../lambda-with-container"
+  codebase    = path.module
```
Comment (baarkerlounger) [inline modules/lambda-example/deploy.tf:3]: Is this just going to zip up everything in this directory?

Comment (baarkerlounger) [reply modules/lambda-example/deploy.tf:3]: I guess my question is will stuff listed in the dockerignore file be excluded from what gets shipped or does the archive zip end up bringing it in?

Comment (acodeninja) [reply modules/lambda-example/deploy.tf:3]: We use the sha256 of the archive to decide if any changes require that the image is rebuilt. The archive itself doesn't get used outside of that. So the .dockerignore is used when the image is built.

I shifted the Dockerfile into the codebase to give better control of dependencies.

Comment (baarkerlounger) [reply modules/lambda-example/deploy.tf:3]: Cool, means you'd have an image rebuild even if ignored file (tests etc) got modified right? But I think that's probably fine.

**Review by baarkerlounger — APPROVED** [review]

File: modules/lambda-example/deploy.tf (hunk 1 above)
Comment (acodeninja) [reply modules/lambda-example/deploy.tf:3]: At the moment, yes. That could be improved, but its probably fine for the moment. One thing I did want to introduce was layer caching to speed up the build time. In testing build time was around 2 minutes without this but could be improved to around 30 seconds per docker image build.

**Pull request merged**